### PR TITLE
Moving from pylint to ruff linter

### PR DESCRIPTION
**Pull request by john-science**
## Description

This PR moves the ARMI project from using the `pylint` linter to using the `ruff` linter.

That means that this PR is big, because we used to have `# pylint: disable=xyz` lines in like every Python file. But that's not necessary with `ruff`.  Not only is `ruff` like 100x faster, it is easier to configure (see the top-level `ruff.toml` file I added).

Also, this PR fixes all `ruff` errors in the repo. This allows me to add a new GitHub Action just to run `ruff` on the code, like we do when we run `black`. This way we can prevent people from adding poorly formatted code to the repo.

Also, `ruff` has the MIT license, which is a big improvement over `pylint`.

---

## Checklist

- [X] This PR has only one purpose or idea.
- [X] Tests have been added/updated to verify that the new/changed code works.

- [X] The code style follows [good practices](https://terrapower.github.io/armi/developer/standards_and_practices.html).
- [X] The commit message(s) follow [good practices](https://terrapower.github.io/armi/developer/tooling.html).

- [X] The [release notes](https://terrapower.github.io/armi/release/index.html) (location `doc/release/0.X.rst`) are up-to-date with any important changes.
- [X] The documentation is still up-to-date in the `doc` folder.
- [X] The dependencies are still up-to-date in `setup.py`.

### CODE REVIEW

**Comment by opotowsky** [comment]
Every time you merge `main` in I feel a little guilt I haven't gotten to it yet. I know it's low priority, but I still expected to be done by now. I promise it's still on my radar, and should work nicely with my schedule in the coming couple of days

**Comment by opotowsky** [comment]
Ok I've started clicking view on files. Finally in progress!

**Review by opotowsky — APPROVED** [review]
Ok this PR looks good! I've also testing things internally and nothing has changed. 

Nice work yet again, @john-science

File: armi/scripts/migration/tests/test_m0_1_6_locationLabels.py [hunk 1]
```diff
@@ -40,7 +39,3 @@ def test_locationLabelMigration(self):
         reader = SettingsReader(newCs)
         reader.readFromStream(converter.apply())
         self.assertEqual(newCs["detailAssemLocationsBOL"][0], "011-012")
-
-
-if __name__ == "__main__":
-    nittest.main()
```
[the remark this answers is not in the document]
Comment (john-science) [reply armi/scripts/migration/tests/test_m0_1_6_locationLabels.py:46]: lol... You can see why I thought maybe the other PR that deleted all these would be okay.  I'm fairly certain they haven't been used in years (doubly confirmed by the fact that like half our test files didn't have them).

**Pull request merged**